src/api/java/com/minecolonies/api/colony/requestsystem/resolver/IRequestResolver.java

```diff
@@ -0,0 +1,129 @@
+package com.minecolonies.api.colony.requestsystem.resolver;
+
+import com.minecolonies.api.colony.requestsystem.IRequestManager;
+import com.minecolonies.api.colony.requestsystem.request.IRequest;
+import com.minecolonies.api.colony.requestsystem.requester.IRequester;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * Used to resolve a request.
+ * In a colony multiple resolvers can exist for a given type R.
+ * The resolver with the highest priority is checked first, then second and so forth.
+ * <p>
+ * The resolver himself is responsible for storing the tokens of requests that he returns
+ *
+ * @param <R> The request type that this resolver can provide.
+ */
+public interface IRequestResolver<R> extends IRequester
```

Comment:
This class is not supposed to be in this PR yet.
Just the Factories part and the IToken part (and there respective implementations)